apps/designer/app/dashboard/project-card.tsx

```diff
@@ -0,0 +1,292 @@
+    setUrl(new URL(getPublishedUrl(domain)));
+  }, [domain]);
+
+  return { url };
+};
+
+const PublishedLink = ({
+  domain,
+  tabIndex,
+}: {
+  domain: string;
+  tabIndex: number;
+}) => {
+  const { url } = usePublishedLink({ domain });
+  return (
+    <Text
+      as="a"
+      href={url?.href}
+      target="_blank"
+      truncate
+      color="hint"
+      tabIndex={tabIndex}
+      css={{
+        "&:not(:hover)": {
+          textDecoration: "none",
+        },
+      }}
+    >
+      {url?.host}
+    </Text>
+  );
+};
+
+const Menu = ({
+  tabIndex,
+  onDelete,
+  onRename,
+  onDuplicate,
+}: {
+  tabIndex: number;
+  onDelete: () => void;
+  onRename: () => void;
+  onDuplicate: () => void;
+}) => {
+  const [isOpen, setIsOpen] = useState(false);
+  return (
+    <DropdownMenu open={isOpen} onOpenChange={setIsOpen}>
+      <DropdownMenuTrigger asChild>
+        <IconButton
+          variant={isOpen ? "active" : "default"}
+          aria-label="Menu Button"
+          tabIndex={tabIndex}
+          css={{ alignSelf: "center" }}
+        >
+          <MenuIcon width={15} height={15} />
+        </IconButton>
+      </DropdownMenuTrigger>
+      <DropdownMenuContent align="end">
+        <DropdownMenuItem onSelect={onDuplicate}>
+          <Text>Duplicate</Text>
+        </DropdownMenuItem>
+        <DropdownMenuItem onSelect={onRename}>
+          <Text>Rename</Text>
+        </DropdownMenuItem>
+        <DropdownMenuItem>
+          <Text>Share</Text>
+        </DropdownMenuItem>
+        <DropdownMenuItem onSelect={onDelete}>
+          <Text>Delete</Text>
+        </DropdownMenuItem>
+      </DropdownMenuContent>
+    </DropdownMenu>
+  );
+};
+
+const useProjectCard = () => {
+  const fetcher = useFetcher();
+  const designerLinkRef = useRef<HTMLAnchorElement>(null);
+
+  // @todo with dialog it can be displayed in the dialog
+  useEffect(() => {
+    if (fetcher.data?.errors) {
+      toast.error(fetcher.data.errors);
+    }
+  }, [fetcher.data]);
+
+  const handleKeyDown = (event: KeyboardEvent<HTMLElement>) => {
+    const elements: Array<HTMLElement> = Array.from(
+      event.currentTarget.querySelectorAll(`[tabIndex='-1']`)
+    );
+    const currentIndex = elements.indexOf(
+      document.activeElement as HTMLElement
+    );
+    switch (event.key) {
+      case "Enter": {
+        designerLinkRef.current?.click();
+        break;
+      }
+      case "ArrowUp":
+      case "ArrowRight": {
+        const nextElement = elements.at(currentIndex + 1) ?? elements[0];
+        nextElement?.focus();
+        break;
+      }
+      case "ArrowDown":
+      case "ArrowLeft": {
+        const nextElement = elements.at(currentIndex - 1) ?? elements[0];
+        nextElement?.focus();
+        break;
+      }
+    }
+  };
+
+  const handleDelete = (projectId: string) => {
+    fetcher.submit(
+      { projectId },
+      { method: "delete", action: "/dashboard/projects/delete" }
```

Comment:
Not a big deal, but a bit strange that we use both `method: "delete"` and `/delete` in path to tell what action to take. Same for `/rename`

If we're trying to follow REST to some extent, we should do something like:

```
/dashboard/projects + PUT = create
/dashboard/projects + POST = edit
/dashboard/projects + DELETE = delete
```

Anyway not a big deal as we probably going to rewrite API at some point.

Reply:
The problem is when there is more than CRUD, e.g. duplicate, you end up needing a separate route for it or alternatively an additional param like "action" or something ...

So there is an option to make all crud operations in /projects and use method  and `ifs` to call all functions from the same action, then have a separate /duplicate and probably also /rename

What would you prefer? also cc @istarkov in relation to trpc what's the way forward?

Reply:
I have no strong preference between:
 - `/dashboard/projects + PUT = create or duplicate depending on input`
 - `/dashboard/projects + PUT = create; /dashboard/projects/duplicate + POST = duplicate`
 
 But if we're going to migrate to trpc, probably no reason to spend a lot of time trying to make it right now